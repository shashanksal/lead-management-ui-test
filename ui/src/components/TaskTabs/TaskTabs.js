import React, { useState } from "react";
import PropTypes from "prop-types";

//Material UI Components
import Tabs from "@mui/material/Tabs";
import Tab from "@mui/material/Tab";
import Box from "@mui/material/Box";
import Typography from "@mui/material/Typography";
import Grid from "@mui/material/Grid";

import { THEME_COLOUR_1 } from "../../utils/constants";

// Styles
import { styled } from "@mui/material/styles";

// Custom components import
import InvitedTab from "../InvitedTab";
import AcceptedTab from "../AcceptedTab";

const CustomTab = styled(props => <Tab disableRipple {...props} />)(({ theme }) => ({
	textTransform: "none",
	minWidth: 0,
	[theme.breakpoints.up("sm")]: {
		minWidth: 0
	},
	border: "1px solid #EEEEEE",
	fontWeight: theme.typography.fontWeightRegular,
	color: "#121212",
	"&.Mui-selected": {
		color: "#121212",
		fontWeight: theme.typography.fontWeightBold
	},
	backgroundColor: "#FFFFFF"
}));

const TabPanel = props => {
	return (
		<Typography component="div" style={{ padding: 8 * 3 }}>
			{props.children}
		</Typography>
	);
};

TabPanel.propTypes = {
	children: PropTypes.node
};

const TaskTabs = () => {
	const [value, setValue] = useState("invited");

	const handleChange = (event, newValue) => {
		console.log("VAl ===", newValue);
		setValue(newValue);
	};

	const tabIndicatorProps = {
		style: { background: THEME_COLOUR_1 }
	};

	return (
		<Grid>
			<Box sx={{ width: "100%" }}>
				<Tabs
					value={value}
					onChange={handleChange}
					aria-label="myapp tabs"
					centered
					TabIndicatorProps={tabIndicatorProps}
					variant={"fullWidth"}
				>
					<CustomTab disableRipple value="invited" label="Invited" />
					<CustomTab disableRipple value="accepted" label="Accepted" />
				</Tabs>
			</Box>
			{value === "invited" && (
				<TabPanel>
					<InvitedTab />
				</TabPanel>
			)}
			{value === "accepted" && (
				<TabPanel>
					<AcceptedTab />
				</TabPanel>
			)}
		</Grid>
	);
};

export default TaskTabs;
