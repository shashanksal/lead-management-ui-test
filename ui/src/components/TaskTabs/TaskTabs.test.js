import React from "react";
import { render } from "@testing-library/react";

import TaskTabs from "./TaskTabs";

describe("TaskTabs", () => {
	it("should match snapshot", () => {
		const { asFragment } = render(<TaskTabs />);
		expect(asFragment()).toMatchSnapshot();
	});
});
