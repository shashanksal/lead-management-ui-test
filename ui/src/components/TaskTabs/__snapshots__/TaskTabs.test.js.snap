// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskTabs should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root css-vj1n65-MuiGrid-root"
  >
    <div
      class="MuiBox-root css-8atqhb"
    >
      <div
        class="MuiTabs-root css-1ujnqem-MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
          style="overflow: hidden; margin-bottom: 0px;"
        >
          <div
            aria-label="myapp tabs"
            class="MuiTabs-flexContainer MuiTabs-centered css-1wf8b0h-MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth Mui-selected css-pabxj7-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="0"
              type="button"
            >
              Invited
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth css-pabxj7-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Accepted
            </button>
          </div>
          <span
            class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
            style="left: 0px; width: 0px; background: rgb(255, 122, 18);"
          />
        </div>
      </div>
    </div>
    <div
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      style="padding-top: 24px;"
    >
      <div
        class="job-card-wrapper MuiBox-root css-0"
      >
        <div
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        >
          No new jobs to show
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
